cmd/note_common.go

```diff
@@ -111,7 +226,12 @@ func createNote(rn string, isMR bool, idNum int, msgs []string, filename string,
 				"merged": "MERGED",
 			}[mr.State]
 
-			body = fmt.Sprintf("\n# This comment is being applied to %s Merge Request %d.", state, idNum)
+			if commit != "" {
+				body = getCommitBody(rn, commit)
+				body += fmt.Sprintf("\n# This comment is being applied to %s Merge Request %d commit %s.\n# Do not delete patch tracking lines that begin with '|'.", state, idNum, commit[:12])
```

Comment:
Forcing the user to pass a SHA of 12 chars seems a bit "overhead", considering the GitLab interface, by default, display all commits using 8 chars instead:

![image](https://user-images.githubusercontent.com/453611/130130495-d3736329-515d-4ef5-a156-e83b4c028e42.png)
![image](https://user-images.githubusercontent.com/453611/130130648-ae8507a5-0459-4dad-a0c9-55d3faae877c.png)

What if we make it, at least, 8 chars?

Reply:
I think 8 chars is not enough.  GitLab in the webUI uses 8 for "pretty" reasons.  The problem is that I've used less than 10 and had git errors indicating that the sha is ambiguous.
For example: 
```

error: short object ID abcd is ambiguous
hint: The candidates are:
hint:   abcd083a1a65 commit 2008-11-25 - sched: convert sched.c from for_each_cpu_mask to for_each_cpu.
hint:   abcd08a6f564 commit 2007-05-09 - SLUB: use check_valid_pointer in kmem_ptr_validate
hint:   abcd3d6fc640 commit 2018-05-04 - net: nixge: Fix error path for obtaining mac address
hint:   abcd7789118c commit 2021-01-14 - [netdrv] ib
```

Reply:
hmmm.. less then 10 chars is already conflicting? That's bad.
But yeah, considering the kernel code.. I don't doubt it :)

What if..... you'll hate me for this: let 8 as default, if git returns "ambiguity" we warn the user to increase the number to at least 12 chars?

I say that because most people using `lab` won't be handling kernel code, but other project far simpler and smaller, so 8 should be a good amount for 90% of the projects out there.

If you prefer, we can let this fix for a follow-up submission, but I think it's important.

Reply:
Good idea!  I'm too kernel-centric in my thinking :)   I've modified the code to error with, for example,

```
[prarit@prarit Test-Lab]$  ~/git-kernel/github/lab/lab mr note 2 --commit abcde
2021/08/19 19:28:41 ERROR: gitlab.go:1803: Cannot find diff for commit abcde.  Verify the commit ID or add more characters to the commit ID.
```

Reply:
I would add "Cannot find diff for commit abcde _or it is ambiguous_".
But that's close enough :)

Reply:
Yeah I wanted something a bit more "human"  to read.  Blame my work on the other tooling lately :)